app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,139 @@
+include ShoppingListHelper
+
+class RecipesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_recipe, only: %i[show edit update destroy]
+  def index
+    @recipes = Recipe.includes(:recipe_foods, :foods).all
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def create
+    @recipe = current_user.recipes.build(recipe_params2)
+
+    if @recipe.save
+      redirect_to @recipe, notice: 'Recipe was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  # app/controllers/recipes_controller.rb
+
+  def generate_shopping_list
+    @recipe = Recipe.find(params[:recipe_id])
+    @inventory = Inventory.find(params[:inventory_id])
+
+    # Add logic to generate the shopping list data here
+    # For example, use your existing logic to calculate missing foods
+    @missing_foods = calculate_missing_foods(@recipe, @inventory)
+    @total_value = calculate_total_value(@missing_foods)
+
+    # Render the shopping list view
+    render 'generate_shopping_list'
+  end
+
+  def calculate_total_value(missing_foods)
+    # Initialize the total value to zero
+    total_value = 0
+
+    # Iterate through the missing_foods array and sum up the prices
+    missing_foods.each do |missing_food|
+      total_value += missing_food[:price]
+    end
+
+    total_value
+  end
+
+  def update_status
+    @recipe = Recipe.find(params[:id])
+    @recipe.public = @recipe.public != true
+    @recipe.save
+    redirect_to @recipe, notice: 'Status updated successfully.'
+  end
+
+  def remove_food
+    @recipe = Recipe.find(params[:id])
+    @recipe_food = @recipe.recipe_foods.find(params[:recipe_food_id])
+
+    if @recipe_food.destroy
+      flash[:notice] = 'Food removed successfully.'
+    else
+      flash[:alert] = 'Error removing food from the recipe.'
+    end
+
+    redirect_to recipe_path(@recipe)
+  end
+
+  def destroy
+    @recipe = Recipe.find(params[:id])
+    return unless @recipe.destroy
+
+    redirect_to recipe_path
+  end
```

Comment:
> * [ ]  Kindly add a delete action to a recipe as shown on the wireframe so that users can delete a recipe they do not want.
> 
> ![recipes](https://user-images.githubusercontent.com/53124548/266776446-6be90985-7532-40b9-8887-356c37f53935.png)
- [ ] As the previous reviewer requested, pleaser add a 'remove' button to each recipe that the user created.